### PR TITLE
sink(ticdc): warning on using pulsar sink

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5928

### What is changed and how it works?
Warning on using Pulsar sink. 

#### Tests 

 - No code

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sink/mq/mq.go [hunk 1]
```diff
@@ -474,6 +474,9 @@ func NewKafkaSaramaSink(ctx context.Context, sinkURI *url.URL,
 func NewPulsarSink(ctx context.Context, sinkURI *url.URL,
 	replicaConfig *config.ReplicaConfig, errCh chan error,
 ) (*mqSink, error) {
+	log.Warn("Pulsar Sink is about to go out of maintenance, " +
+		"it has not been production tested, and there is no support for new features. " +
+		"Please try not to use it, it will stop being supported in the new version.")
```
Comment (leoppro) [inline cdc/sink/mq/mq.go:479]:
```suggestion
	log.Warn("is not recommended for production use. ")
```

**Review by leoppro — APPROVED** [review]
LGTM

**Review by overvenus — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Comment by hi-rustin** [comment]
I've done some local tests. It works.
image

**Review by hi-rustin — COMMENTED** [review]
self check

**Pull request merged**